spec/models/song_spec.rb

```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+describe Song do
+  it { should validate_presence_of :title }
+  it { should validate_presence_of :artist }
+  it { should validate_presence_of :album }
+  it { should validate_presence_of :duration }
```

Comment:
All these validations are spot on and really doesn't need much more.

But ... can you think of any other validation(s) you might add, say on `:duration` that could be useful?